src/libstd/path.rs

```diff
@@ -0,0 +1,2415 @@
+    pub fn from_str(s: &str) -> &Path {
+        Path::from_os_str(OsStr::from_str(s))
+    }
+
+    /// Directly wrap an `OsStr` slice as a `Path` slice.
+    ///
+    /// This is a cost-free conversion.
+    pub fn from_os_str(s: &OsStr) -> &Path {
+        unsafe { mem::transmute(s) }
+    }
+
+    /// Extract the underlying `OsStr` slice.
+    pub fn as_os_str(&self) -> &OsStr {
+        &self.inner
+    }
+
+    /// Yield a `&str` slice if the `Path` is valid unicode.
+    ///
+    /// This conversion may entail doing a check for UTF-8 validity.
+    pub fn to_str(&self) -> Option<&str> {
+        self.inner.to_str()
+    }
+
+    /// Convert a `Path` to a `CowString`.
+    ///
+    /// Any non-Unicode sequences are replaced with U+FFFD REPLACEMENT CHARACTER.
+    pub fn to_string_lossy(&self) -> CowString {
+        self.inner.to_string_lossy()
+    }
+
+    /// Convert a `Path` to an owned `PathBuf`.
+    pub fn to_path_buf(&self) -> PathBuf {
+        PathBuf::from_path(self)
+    }
+
+    /// A path is *absolute* if it is indepedent of the current directory.
+    ///
+    /// * On Unix, a path is absolute if it starts with the root, so
+    /// `is_absolute` and `has_root` are equivalent.
+    ///
+    /// * On Windows, a path is absolute if it has a prefix and starts with the
+    /// root: `c:\windows` is absolute, while `c:temp` and `\temp` are not. In
+    /// other words, `path.is_absolute() == path.prefix().is_some() && path.has_root()`.
+    pub fn is_absolute(&self) -> bool {
+        self.has_root() &&
+            (cfg!(unix) || self.prefix().is_some())
+    }
+
+    /// A path is *relative* if it is not absolute.
+    pub fn is_relative(&self) -> bool {
+        !self.is_absolute()
+    }
+
+    /// Returns the *prefix* of a path, if any.
+    ///
+    /// Prefixes are relevant only for Windows paths, and consist of volumes
+    /// like `C:`, UNC prefixes like `\\server`, and others described in more
+    /// detail in `std::os::windows::PathExt`.
+    pub fn prefix(&self) -> Option<&Path> {
+        let iter = self.components();
+        optional_path(unsafe {
+            Path::from_u8_slice(
+                &self.as_u8_slice()[.. iter.prefix_remaining()])
+        })
+    }
+
+    /// A path has a root if the body of the path begins with the directory separator.
+    ///
+    /// * On Unix, a path hss a root if it begins with `/`.
+    ///
+    /// * On Windows, a path has a root if it:
+    ///     * has no prefix and begins with a separator, e.g. `\\windows`
+    ///     * has a prefix followed by a separator, e.g. `c:\windows` but not `c:windows`
+    ///     * has any non-disk prefix, e.g. `\\server\share`
+    pub fn has_root(&self) -> bool {
+         self.components().has_root()
+    }
+
+    /// The path without its final component.
+    ///
+    /// Does nothing, returnning `None` if the path consists of just a prefix
+    /// and/or root directory reference.
+    pub fn parent(&self) -> Option<&Path> {
```

Comment:
Does this mean that `"a/b".parent() == "a"`?

(may be nice to have some quick examples)